### PR TITLE
Adding default permissions except view to run tests on django2.

### PR DESCRIPTION
**Pull request by awais786**
https://openedx.atlassian.net/browse/BOM-1411
Fixing view permission.

As per [django](https://docs.djangoproject.com/en/2.2/ref/models/options/#default-permissions).
Defaults to ('add', 'change', 'delete', 'view'). You may customize this list, for example, by setting this to an empty list if your app doesn’t require any of the default permissions. It must be specified on the model before the model is created by migrate in order to prevent any omitted permissions from being created.

My understanding is `'add', 'change', 'delete'` are already there in django1.11, they included the `view permission` in django2 which conflicts with our custom view migration. Two view objects created and gives integrity error. 

To avoid this issue I just removed the view permission from default list other permissions stays there.

I don't think there is any permission effect on discovery. All tests are green. 
@jmbowman your thoughts on it.

### CODE REVIEW

**Comment by awais786** [comment]
I have added this test to verify permission exists in db.
https://github.com/edx/course-discovery/blob/master/course_discovery/apps/catalogs/tests/test_models.py#L102

**Comment by awais786** [comment]
@dianakhuang this fix is working fine.

**Comment by awais786** [comment]
@Dillon-Dumesnil @mikix Please review.

**Review by jmbowman — APPROVED** [review]
Looks good to me.

Also, looks like it's time to remove `allow_failures` in Travis for the Django 2.x tests, since they're passing now.

File: course_discovery/apps/catalogs/models.py [hunk 1]
```diff
@@ -125,3 +125,4 @@ class Meta(TimeStampedModel.Meta):
         permissions = (
             ('view_catalog', 'Can view catalog'),
         )
+        default_permissions = ('add', 'change', 'delete',)
```
Comment (mikix) [inline course_discovery/apps/catalogs/models.py:130]: It'd be nice if this had a comment explaining why we do this (historical reasons, etc). Would look weird to someone coming at this a year from now.

**Review by mikix — APPROVED** [review]

**Comment by dianakhuang** [comment]
@awais786 it looks like Travis tests are passing: https://travis-ci.org/github/edx/course-discovery/builds/669246558

I can merge this branch once the status gets reported back to Github.

**Comment by awais786** [comment]
> @awais786 it looks like Travis tests are passing: https://travis-ci.org/github/edx/course-discovery/builds/669246558
> 
> I can merge this branch once the status gets reported back to Github.

I am waiting for last 1 hr but no update. I can't see enabled merge button due to this.

**Comment by mikix** [comment]
Travis has been not reporting statuses lately...?  It still does the work and can be manually merged by disco admins. Bug us in the publisher eng channel.

**Pull request merged**